core/Pos/Binary/Class/Core.hs

```diff
@@ -388,22 +432,23 @@ setTag = 258
 encodeSetTag :: E.Encoding
 encodeSetTag = E.encodeTag setTag
 
-decodeSetTag :: D.Decoder s ()
+decodeSetTag :: Decoder s ()
 decodeSetTag = do
-    t <- D.decodeTagCanonical
-    when (t /= setTag) $ fail ("decodeSetTag: this doesn't appear to be a Set. Found tag: " <> show t)
+    t <- toDecoder D.decodeTagCanonical
+    when (t /= setTag) $
+        fail ("decodeSetTag: this doesn't appear to be a Set. Found tag: " <> show t)
 
-decodeSetSkel :: (Ord a, Bi a) => ([a] -> c) -> D.Decoder s c
+decodeSetSkel :: (Ord a, Bi a) => ([a] -> c) -> Decoder s c
```

Comment:
This function checks the order of items. I think we don't need this check when `dcNoCheck` is `True`.